test/conformance_test.jl

```diff
@@ -19,8 +19,6 @@ function test_Group_interface(G::Group)
                     @test first(iterate(G, s)) isa eltype(G)
                     @test isone(first(G))
                 end
-            else
-                @test isfinite(G) == false
```

Comment:
I think this testset hasn't been finished:
```julia
elseif IS isa Base.IsInfinite
     ...
else
    @test IS isa Base.SizeUnknown
    @test_throws ArgumentError isfinite(G)
end
```

Reply:
That also makes no sense. IS is derived only from the type of G. But there are types which have both finit and infinite instances. So isfinite will not throw an exception in general.

Reply:
then surround `@test_throws` by `try` statement and test first whether `isfinite` returns a `Bool`

Reply:
I've split this PR into three; this original PR retains the iterator test changes. I've now adjusted them along the lines of what you wrote, but in a way that ensures the tests for finite groups are performed solely based on whether `isfinite(G) == true`, regardless of what `Base.IteratorSize(typeof(G))` gives.